torch/csrc/jit/script/parser.h

```diff
@@ -297,12 +285,17 @@ struct Parser {
         return parseWhile();
       case TK_GLOBAL: {
         auto range = L.next().range;
-        std::vector<TreeRef> idents;
-        do {
-          idents.push_back(parseIdent());
-        } while (L.nextIf(','));
-        expectEndOfLine();
-        return c(TK_GLOBAL, range, std::move(idents));
+        auto idents = parseList(TK_NOTHING, ',', TK_NOTHING, &Parser::parseIdent);
+        if (idents.empty())
+          throw ErrorReport(range) << "global statement needs a list of variable names";
```

Comment:
How can this happen? Assuming there are no TK_NOTHING tokens in the stream, then there is at least one element in this list or an exception is thrown.

Reply:
Right. Now that I read the body again, it uses a `do ... while` instead of `while`, so it has to be non-empty when end is `TK_NOTHING`.